app/uk/gov/hmrc/gform/des/DesConnector.scala

```diff
@@ -55,14 +57,12 @@ class DesConnector(wSHttp: WSHttp, baseUrl: String, desConfig: DesConnectorConfi
     Logger.info(
       s"Des lookup, Tax Periods: '$idType, $idNumber, $regimeType', ${loggingHelpers.cleanHeaderCarrierHeader(hc)}")
     val value = s"$baseUrl${desConfig.basePath}/enterprise/obligation-data/$idType/$idNumber/$regimeType?status=O"
-    val response = wSHttp.GET[JsValue](value)
+    val response = wSHttp.GET[HttpResponse](value)
     response.map { i =>
-      val checkObligation = i.validate[Obligation]
-      val checkNoPeriods = i.validate[NoPeriods]
-      (checkObligation, checkNoPeriods) match {
-        case (JsSuccess(o, _), _) => o
-        case (_, JsSuccess(_, _)) => Obligation(List())
-        case _                    => throw new JsValidationException("GET", value, Obligation.getClass, "Incorrect response from api")
+      i.status match {
+        case Status.OK        => i.json.validate[Obligation].get
+        case Status.NOT_FOUND => Obligation(List())
+        case _                => throw new Upstream5xxResponse("Misc Failure", 500, 500)
```

Comment:
You are saying here that DES has returned a 500, but there are lots of things it could have returned